tests/e2e/steps/steps_def.go

```diff
@@ -752,6 +756,46 @@ func (step *Step) setUpSecret(res Resource, templateFile, name, namespace, crTyp
 	return nil
 }
 
+func (step *Step) setUpTempSecret(res Resource, templateFile, name, namespace, crType string) error {
+	// find which map to use for secret values
+	mapValues, err := determineMap(crType)
+	if err != nil {
+		return err
+	}
+
+	// read the template into memory
+	fileContent, err := os.ReadFile(templateFile)
+	if err != nil {
+		fmt.Println("Error reading file:", err)
+		return err
+	}
+
+	// Convert the file content to a string
+	fileString := string(fileContent)
+
+	// Replace all fields in temporary (in memory) string
+	for key := range mapValues {
+		fileString = strings.ReplaceAll(fileString, key, os.Getenv(mapValues[key]))
+	}
+
+	// if secret exists- delete it
+	if secretExists(namespace, name) {
+		err := execCommand("kubectl", "delete", "secret", "-n", namespace, name)
+		if err != nil {
+			return fmt.Errorf("failed to delete secret: %s", err.Error())
+		}
+	}
+
+	// create new secret
+	fileArg := "--from-literal=config=" + fileString
+	err = execCommand("kubectl", "create", "secret", "generic", "-n", namespace, name, fileArg)
```

Comment:
Have you tested this with special shell characters in fileString, e.g. dollar signs and exclamation marks.